server/src/bigquery/BigQueryContext.ts

```diff
@@ -1,43 +1,42 @@
 import { AnalyzeResponse__Output } from '@fivetrandevelopers/zetasql/lib/types/zetasql/local_service/AnalyzeResponse';
-import { err, ok, Result } from 'neverthrow';
+import { Err, err, Ok, ok, Result } from 'neverthrow';
 import { TextDocument } from 'vscode-languageserver-textdocument';
-import { DbtProfileCreator, DbtProfileError, DbtProfileSuccess } from '../DbtProfileCreator';
+import { DbtProfileCreator } from '../DbtProfileCreator';
 import { DestinationDefinition } from '../DestinationDefinition';
 import { SchemaTracker } from '../SchemaTracker';
 import { YamlParser } from '../YamlParser';
 import { ZetaSqlWrapper } from '../ZetaSqlWrapper';
 import { BigQueryClient } from './BigQueryClient';
 
-interface PresentBigQueryContext {
-  schemaTracker: SchemaTracker;
-  destinationDefinition: DestinationDefinition;
-  zetaSqlWrapper: ZetaSqlWrapper;
+export interface ContextInfo {
+  type?: string;
+  method?: string;
 }
 
-export class BigQueryContext {
-  private static CONTEXT_NOT_INITIALIZED_ERROR = 'Context is not initialized.';
-
-  private dbtProfileCreator: DbtProfileCreator;
-
-  private present?: boolean;
-  private presentBigQueryContext?: PresentBigQueryContext;
-
-  public constructor(private yamlParser: YamlParser) {
-    this.dbtProfileCreator = new DbtProfileCreator(this.yamlParser);
-  }
+export interface ErrorContextInfo extends ContextInfo {
+  error: string;
+}
 
-  public async initialize(): Promise<Result<DbtProfileSuccess, DbtProfileError>> {
+export class BigQueryContext {
+  private constructor(
+    public contextInfo: ContextInfo,
+    public schemaTracker: SchemaTracker,
+    public destinationDefinition: DestinationDefinition,
+    public zetaSqlWrapper: ZetaSqlWrapper,
+  ) {}
+
+  public static async createContext(yamlParser: YamlParser): Promise<Result<BigQueryContext, ErrorContextInfo>> {
```

Comment:
It will lead to repeated actions. Moreover positive result of first function will not guarantee that context could be created. Some error could occur on ZetaSQL initialization.